src/main/java/no/ndla/taxonomy/rest/v1/CrudControllerWithMetadata.java

```diff
@@ -24,26 +27,32 @@
 import java.net.URI;
 
 public abstract class CrudControllerWithMetadata<T extends DomainEntity> extends CrudController<T> {
-    private final MetadataService metadataService;
-
     protected CrudControllerWithMetadata(TaxonomyRepository<T> repository,
-            CachedUrlUpdaterService cachedUrlUpdaterService, MetadataService metadataService) {
+            CachedUrlUpdaterService cachedUrlUpdaterService) {
         super(repository, cachedUrlUpdaterService);
-
-        this.metadataService = metadataService;
     }
 
     @GetMapping("/{id}/metadata")
     @Operation(summary = "Gets metadata for entity")
     public MetadataDto getMetadata(@PathVariable("id") URI id) {
-        return metadataService.getMetadataByPublicId(id);
+        var entity = repository.findByPublicId(id);
+        if (entity instanceof EntityWithMetadata em) {
+            return new MetadataDto(em.getMetadata());
+        }
+        throw new NotFoundException("Entity", id);
     }
 
     @PutMapping(path = "/{id}/metadata")
     @PreAuthorize("hasAuthority('TAXONOMY_WRITE')")
     @Operation(summary = "Updates metadata for entity", security = { @SecurityRequirement(name = "oauth") })
     public MetadataDto putMetadata(@PathVariable("id") URI id, @RequestBody MetadataDto entityToUpdate)
             throws InvalidDataException {
-        return metadataService.updateMetadataByPublicId(id, entityToUpdate);
+        var entity = repository.findByPublicId(id);
+        if (entity instanceof EntityWithMetadata em) {
+            var metadata = new Metadata(entityToUpdate);
```

Comment:
Må nok inn med litt kode for å merge data, i og med at eksisterende løsning støtter delvis oppdatering.